ModelFarm/src/test/java/io/deephaven/modelfarm/TestModelFarm.java

```diff
@@ -51,32 +54,33 @@ public void testModelFarmNoWorkShutdown() throws Exception {
      * {@link ModelFarmBase#awaitTermination}, but not {@link ModelFarmBase#terminate()}, are called.
      */
     public void testModelFarmBusyShutdownNoTerminate() throws Exception {
-        final long workerSleepMs = 1000 * 3;
         final int nModelFarmThreads = 2;
         final int nKeys = nModelFarmThreads / 2;
+        final CountDownLatch latchThreadStart = new CountDownLatch(1);
+        final CountDownLatch latchThreadComplete = new CountDownLatch(1);
         final AtomicInteger successCounter = new AtomicInteger(0);
         final ModelFarmTick<Long, MutableLong, RowDataManager<Long, MutableLong>> modelFarmTick =
-                getModelFarmTick(nModelFarmThreads, nKeys, workerSleepMs, successCounter);
+                getModelFarmTick(nModelFarmThreads, nKeys, latchThreadStart, latchThreadComplete, successCounter);
 
         Require.eq(modelFarmTick.getState(), "modelFarmTick.getState()", ModelFarmBase.State.WAITING);
 
         modelFarmTick.start();
         Require.eq(modelFarmTick.getState(), "modelFarmTick.getState()", ModelFarmBase.State.RUNNING);
 
-        Thread.sleep(100); // Give the model farm threads a bit of time to start up and get far enough to pick up some
-                           // work.
+        // wait on the threads to start up and get into the queue (latchThreadStart=1 => wait on a thread to come up)
+        latchThreadStart.await();
 
         Require.eq(modelFarmTick.getState(), "modelFarmTick.getState()", ModelFarmBase.State.RUNNING);
 
-        // Shut down the model farm. Since the worker threads will sleep for (workerSleepMs) milliseconds,
+        // Shut down the model farm. Since the worker threads are blocked,
         // the model farm will still be busy -- so its state should be SHUTDOWN, not TERMINATED.
         modelFarmTick.shutdown();
         Require.eq(modelFarmTick.getState(), "modelFarmTick.getState()", ModelFarmBase.State.SHUTDOWN);
 
         // Test awaitTermination(), without calling terminate().
-        // This should return true, we are waiting for (workerSleepMs * 2) milliseconds but all work should be finished
-        // within (workerSleepMs) milliseconds.
-        boolean terminated = modelFarmTick.awaitTermination(workerSleepMs * 2, TimeUnit.MILLISECONDS);
+        // This should return true, since threads are finished.
+        latchThreadComplete.countDown();
```

Comment:
This doesn't make sense to me. You're making it so the ModelFarm shutdown won't stop the threads until some external latch is counted down. That seems contrary to what you would want.

Reply:
The latch is used to control when the ModelFarm threads are complete.  This allows me to test shutdown vs terminate.  Shutdown is a soft kill that lets work complete.  Terminate is a hard kill that kills running threads.  It is better than the multiple sleeps in the prior version.